File: PolicyPlus/EditPolMultiStringData.vb
Public Class EditPolMultiStringData
    Private Sub EditPolMultiStringData_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.Escape Then DialogResult = DialogResult.Cancel
    End Sub
    Public Function PresentDialog(ValueName As String, InitialData As String()) As DialogResult
        TextName.Text = ValueName
        TextData.Lines = InitialData
        TextData.Select()
        Return ShowDialog()
    End Function
End Class

File: PolicyPlus/EditPolMultiStringData.cs
using System.Windows.Forms;

namespace PolicyPlus
{
    public partial class EditPolMultiStringData
    {
        public EditPolMultiStringData()
        {
            InitializeComponent();
        }
        private void EditPolMultiStringData_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
                DialogResult = DialogResult.Cancel;
        }
        public DialogResult PresentDialog(string ValueName, string[] InitialData)
        {
            TextName.Text = ValueName;
            TextData.Lines = InitialData;
            TextData.Select();
            return ShowDialog();
        }
    }
}